code/modules/integrated_electronics/core/analyzer.dm

```diff
@@ -8,6 +8,9 @@
 
 /obj/item/integrated_electronics/analyzer/afterattack(var/atom/A, var/mob/living/user)
 	. = ..()
+	if(istype(A, /obj/item/implant/integrated_circuit)) //snoooooowflake code
```

Comment:
Any reason at all that this snowflake code can't be replaced with a simple oneline `A = A.get_circuit()`, with `get_circuit()` being a proc in `/atom` that returns itself by default?